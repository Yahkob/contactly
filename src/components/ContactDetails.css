.contactDetail {
  margin: 10px auto;
  width: 400px;
  position: relative;
  background: white;
  padding: 20px;
  line-height: 23px;
}

.contactIcon {
  background: url("../assets/user-icon.png") 0px 0px no-repeat;
  background-size: 50px;
  width: 50px;
  height: 50px;
  display: inline-block;
}

.contactInfo {
  display: inline-block;
  margin-left: 20px;
}

.contactDetail a {
  text-decoration: none;
  color: black;
  position: absolute;
  left: -70px;
  top: 49px;
}