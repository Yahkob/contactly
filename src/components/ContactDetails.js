import React from 'react';
import Proptypes from 'prop-types';
import { Link } from "react-router-dom";
import './ContactDetails.css';

const ContactDetails = ({ contact }) => {
  return (
    <div className="contactDetail">
      <Link to="/">â—€ Back</Link>
      <div className="contactIcon" />
      <div className="contactInfo">
        <div className="name">
          {contact.name}
        </div>
        <div className="number">
          {contact.number}
        </div>
        <div className="email">
          {contact.email}
        </div>
      </div>
    </div>
  );
};

ContactDetails.Proptypes = {
  contact: Proptypes.object
};

export default ContactDetails;
