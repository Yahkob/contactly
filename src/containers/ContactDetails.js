import { connect } from 'react-redux';
import { fetchContactsList } from '../actions';
import React, { Component } from 'react';
import ContactDetails from '../components/ContactDetails';
import PropTypes from 'prop-types';

const mapDispatchToProps = dispatch => ({
  fetchContactsList: () => dispatch(fetchContactsList())
});

const mapStateToProps = state => ({
  list: state.contacts,
  getContact: contactId => state.contacts.find(contact => contact.id === contactId)  
});

class ContactDetailsContainer extends Component {
  componentDidMount() {
    if (!this.props.list.length) {
      this.props.fetchContactsList();
    }
  }
  render() {
    const contact = this.props.list && this.props.getContact(this.props.match.params.id);
    return (
      <div>
        {contact && <ContactDetails contact={contact} />}
      </div>
    );
  }
}

ContactDetailsContainer.PropTypes = {
  fetchContactsList: PropTypes.func.isRequired,
  getContact: PropTypes.func.isRequired,
  list: PropTypes.array
};    

export default connect(mapStateToProps, mapDispatchToProps)(ContactDetailsContainer);
