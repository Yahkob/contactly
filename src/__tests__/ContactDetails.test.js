import React from 'react';
import ContactDetails from '../components/ContactDetails.js';
import renderer from 'react-test-renderer';
import MockData from '../contacts.json';
import { MemoryRouter } from "react-router-dom";

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ContactDetails contact={MockData[0]} />
      </MemoryRouter>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
