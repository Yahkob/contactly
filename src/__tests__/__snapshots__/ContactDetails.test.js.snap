// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="contactDetail"
>
  <a
    href="/"
    onClick={[Function]}
  >
    â—€ Back
  </a>
  <div
    className="contactIcon"
  />
  <div
    className="contactInfo"
  >
    <div
      className="name"
    >
      Batman
    </div>
    <div
      className="number"
    >
      555-123-4567
    </div>
    <div
      className="email"
    >
      bruce@wayneenterprises.com
    </div>
  </div>
</div>
`;
